lib/Doctrine/ORM/ORMInvalidArgumentException.php

```diff
@@ -104,96 +76,54 @@ static function (array $newEntityWithAssociation): string {
         );
     }
 
-    /**
-     * @param object $entry
-     * @psalm-param array<string, string> $associationMapping
-     *
-     * @return ORMInvalidArgumentException
-     */
-    public static function newEntityFoundThroughRelationship(array $associationMapping, $entry)
+    /** @psalm-param array<string, string> $associationMapping */
+    public static function newEntityFoundThroughRelationship(array $associationMapping, object $entry): ORMInvalidArgumentException
     {
         return new self(self::newEntityFoundThroughRelationshipMessage($associationMapping, $entry));
     }
 
-    /**
-     * @param object $entry
-     * @psalm-param array<string, string> $assoc
-     *
-     * @return ORMInvalidArgumentException
-     */
-    public static function detachedEntityFoundThroughRelationship(array $assoc, $entry)
+    /** @psalm-param array<string, string> $assoc */
```

Comment:
Probably something we should fix on a lower branch: The method has a pretty strong opinion on which keys should be defined on that array. It might be worth either documenting the shape or splitting `$assoc` into three (?) parameters.